pkg/xds/sync/dataplane_sync_tracker.go

```diff
@@ -22,17 +22,30 @@ type NewDataplaneWatchdogFunc func(dataplaneId core_model.ResourceKey, streamId
 func NewDataplaneSyncTracker(factoryFunc NewDataplaneWatchdogFunc) envoy_xds.Callbacks {
 	return &dataplaneSyncTracker{
 		newDataplaneWatchdog: factoryFunc,
-		streams:              make(map[int64]context.CancelFunc),
+		streamsAssociation:   make(map[int64]core_model.ResourceKey),
+		dpStreams:            make(map[core_model.ResourceKey]streams),
 	}
 }
 
 var _ envoy_xds.Callbacks = &dataplaneSyncTracker{}
 
+type streams struct {
+	watchdogCancel context.CancelFunc
+	activeStreams  map[int64]bool
```

Comment:
I thought we already have type alias for `int64`. Could it be used here as well?

Reply:
I moved the sds `streamID` to core_xds and reused here